syntax = "proto3";

message WeatherRequest {
    City coordinates = 1;
}

message City {
    string name = 1;
}

message WeatherResponse {
    Temperature temperature = 1;
    Wind wind = 2;
    Humidity humidity = 3;
}

message Temperature {
    float degrees = 1;
    Units units = 2;
    enum Units {
        FAHRENHEIT = 0;
        CELSUIS = 1;
        KELVIN = 2;
    }
}

message Wind {
    Speed speed = 1;
    float direction = 2;
}

message Speed {
    float value = 1;
    Units units = 2;

    enum Units {
        MPH = 0;
        MPS = 1;
        KNOTS = 2;
        KMH = 3;
    }
}

message Humidity {
    float value = 1;
}
